python/ray/autoscaler/_private/_azure/node_provider.py

```diff
@@ -65,6 +66,17 @@ class AzureNodeProvider(NodeProvider):
 
     def __init__(self, provider_config, cluster_name):
         NodeProvider.__init__(self, provider_config, cluster_name)
+        if "subscription_id" not in provider_config:
```

Comment:
Let's do this the exact same way both here and in `python/ray/autoscaler/_private/_azure/config.py` (in `_configure_resource_group`).

If we do this after L80 we can avoid polluting the actual `provider_config`, e.g., a new addition @ L81:

```python
if subscription_id is None:
        # Get subscription from logged in azure profile
        # if it isn't provided in the provider_config
        # so operations like `get-head-ip` will work
        subscription_id = get_cli_profile().get_subscription_id()
        logger.info(
                "subscription_id not found in provider config, falling back "
                f"to subscription_id from the logged in azure profile: {subscription_id}"
            )
```

Reply:
let me check to see if the value is pulled out of the provider config dictionary later and if it isn't I'll update with this suggestion

Reply:
removed the line to re-add it to the dictionary and tested various ray command (up, down, status, get-head-ip) and all work as expected.